aspnetcore/fundamentals/configuration.md

```diff
@@ -5,162 +5,981 @@ ms.author: riande
+           _optionsAccessor = optionsAccessor.Value;
+       }
+
+       public IActionResult Index()
+       {
+           var option1 = _optionsAccessor.Option1;
+           var option2 = _optionsAccessor.Option2;
+           return Content($"option1 = {option1}, option2 = {option2}");
+       }
+   }
+
+   ````
+
+With the following *appsettings.json* file:
+
+[!code[Main](configuration/sample/src/UsingOptions/appsettings1.json)]
+
+<!-- literal_block {"xml:space": "preserve", "language": "json", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/fundamentals/configuration/sample/src/UsingOptions/appsettings1.json", "highlight_args": {"linenostart": 1}, "names": []} -->
+
+````json
+
+   {
+removre me
+     "option1": "value1_from_json",
+     "option2": 2
+   }
+
+   ````
+
+The `HomeController.Index` method returns `option1 = value1_from_json, option2 = 2`.
+
+In the following code, a second `IConfigureOptions<TOptions>` service is added to the service container. It uses a delegate to configure the binding with `MyOptions`.
+
+[!code-csharp[Main](configuration/sample/src/UsingOptions/Startup2.cs?name=snippet1)]
+<!--
+http://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/Extensions/Options/IConfigureOptions-TOptions/index.html.md#Microsoft.Extensions.Options.IConfigureOptions<TOptions>
+-->
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/fundamentals/configuration/sample/src/UsingOptions/Startup2.cs", "highlight_args": {"linenostart": 1, "hl_lines": [9, 10, 11, 12, 13]}, "names": []} -->
+
+````c#
+
+   public void ConfigureServices(IServiceCollection services)
+   {
+// rm
+       // Adds services required for using options.
+       services.AddOptions();
+
+       // Register the ConfigurationBuilder instance which MyOptions binds against.
+       services.Configure<MyOptions>(Configuration);
+
+       // Registers the following lambda used to configure options.
+       services.Configure<MyOptions>( myOptions =>
+       {
+           myOptions.Option1 = "value1_from_action";
+       });
+
+       // Add framework services.
+       services.AddMvc();
+   }
+
+   ````
+
+You can add multiple configuration providers. Configuration providers are available in NuGet packages. They are applied in order they are registered. Each call to `Configure<TOptions>]` adds an `IConfigureOptions<TOptions>` service to the service container. In the example above, the values of `Option1` and `Option2` are both specified in *appsettings.json*, but the value of `Option1` is overridden by the configured delegate in the highlighted code above. When more than one configuration service is enabled, the last configuration source specified “wins”. With the code above, the `HomeController.Index` method returns `option1 = value1_from_action, option2 = 2`.
```

Comment:
Fourth sentence should probably start a new paragraph to keep it clear that Configuration providers and `IConfigureOptions<T>` registrations are very independent things.